src/Service.php

```diff
@@ -7,135 +7,287 @@
+				$this->coveredComponents[] = ["dir" => $dirSetup['componentDir'], "section" => $key];
 			}
-			
-			list($class, $method) = explode('::', $method);
-			
-			$urls = array_merge($urls, (new $class)->$method());
 		}
+		return $this->robotLoader;
+	}
 
-		return $urls;
+
+	/**
+	 * Pouze getter, vraci informace o selhani nastaveni konfigurace
+	 */
+	public function getSkippedForMissingConfiguration() {
+		return $this->skippedForMissingConfiguration;
 	}
 
-	protected function getMethods() : array
-	{
-		$methods = [];
+
+	/**
+	 * Metoda vyhleda vsechny soubory podle zadane masky a nastavi je do pole
+	 * @throws \ReflectionException
+	 */
+	protected function findMethodsToCover(): void {
+
+		if (!empty($this->methodsToCover)) {
+			return;
+		}
+
 		foreach ($this->getRobotLoader()->getIndexedClasses() as $_className => $_classFile) {
-			if (! Strings::startsWith($_classFile, $this->config['presenterDir'] . '/')) {
+
+			// kontrola jestli se jedna o neco co ma byt pokryto testy
+			$notFound = true;
+			$section = '';
+			foreach ($this->coveredComponents as $key => $componentDir) {
+				// kontrola jestli dany soubor je soucasti souboru, ktere obsahuji implementaci pro kterou by mel existovat test.
+				if (! Strings::startsWith($_classFile, $componentDir['dir'] . '/')) {
+					continue;
+				}
+				// nalezen, nastavime notFound na false a ukoncime iterovani
+				$notFound = false;
+				$section = $componentDir['section'];
+				break;
+			}
+
+			// pokud jsme dany soubor nenasli, budeme pokracovat dalsi iteraci
+			if ($notFound) {
 				continue;
 			}
 
-			if (! Strings::endsWith($_classFile, 'Presenter.php')) {
+			$mask = $this->config['componentCoverage'][$section]['fileMask'];
+
+			if (! Strings::endsWith($_classFile, $mask)) {
 				continue;
 			}
 
 			require_once $_classFile;
 
 			$_presenterReflection = new \ReflectionClass($_className);
 
+			// nechceme zpracovavat abstraktni tridy -> continue
+			if ($_presenterReflection->isAbstract()) {
+				continue;
+			}
+
+			// maska metody, chceme zpracovat jenom metody ktere maji danou masku
+			$methodMask = null;
+			if (isset($this->config['componentCoverage'][$section]['methodMask'])) {
+				$methodMask = $this->config['componentCoverage'][$section]['methodMask'];
+			}
+
 			foreach ($_presenterReflection->getMethods() as $_presenterMethodReflection) {
-				if (! static::isMethodToTest($_presenterMethodReflection->getName())) {
+				// testy na abstraktni metody nemaji smysl -> continue
+				if ($_presenterMethodReflection->isAbstract()) {
 					continue;
 				}
 
-				$methods[] = $this->getTestClassAndMethod($_presenterReflection->getName(), $_presenterMethodReflection->getName());
+				if (! static::isMethodToTest($_presenterMethodReflection->getName(), $methodMask)) {
+					continue;
+				}
+
+				// potrebujeme ziskat cestu k souboru v mistni slozce, ta odpovida namespace -> smazeme to co je pred namespacem
```

Comment:
Dejme ke všem těmhle stringovým operacím příklad z čeho na co to převádíme, ať se nám pak líp dohledávají případné chyby.

Reply:
Přidal jsem příklady toho k jakým transformacím stringů dochází